hyper/http11/connection.py

```diff
@@ -137,6 +144,33 @@ def connect(self):
 
         return
 
+    @classmethod
```

Comment:
The reason I've made it a classmethod is to ensure that the `HTTP11Connection` instance, created to connect to a proxy, is not usable/alterable anymore, because its socket is moved to an original connection instance, which picks it up.

Maybe a staticmethod will be better? I'm afraid that an instance method doesn't incapsulate this fragile socket transition properly